app/src/main/java/org/p2p/wallet/solend/ui/earn/SolendEarnPresenter.kt

```diff
@@ -2,46 +2,45 @@ package org.p2p.wallet.solend.ui.earn
 
 import android.content.Context
 import org.p2p.wallet.common.mvp.BasePresenter
-import org.p2p.wallet.solend.model.SolendDepositToken
+import org.p2p.wallet.solend.interactor.SolendDepositsInteractor
 import org.p2p.wallet.utils.getErrorMessage
 import timber.log.Timber
-import java.math.BigDecimal
-import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 
+private val COLLATERAL_ACCOUNTS = listOf("SOL", "USDT", "USDC", "BTC", "ETH")
+
 class SolendEarnPresenter(
-    private val context: Context
+    private val context: Context,
+    private val solendDepositsInteractor: SolendDepositsInteractor
 ) : BasePresenter<SolendEarnContract.View>(), SolendEarnContract.Presenter {
 
     override fun load() {
         view?.showLoading(isLoading = true)
         launch {
             try {
-                // TODO: do request and show real data
-                delay(1000L)
-                val tokens = listOf(
-                    SolendDepositToken.Active(
-                        "Solana",
-                        "SOL",
-                        "https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/" +
-                            "So11111111111111111111111111111111111111112/logo.png",
-                        3.05f, BigDecimal.valueOf(100)
-                    ),
-                    SolendDepositToken.Inactive(
-                        "Tether USD",
-                        "USDT",
-                        "https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/" +
-                            "Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB/logo.svg",
-                        2.05f
-                    )
-                )
-                view?.showTokens(tokens)
+                val deposits = solendDepositsInteractor.getAllDeposits(COLLATERAL_ACCOUNTS)
+                view?.showDeposits(deposits)
             } catch (e: Throwable) {
                 Timber.e(e, "Error fetching available deposit tokens")
-                view?.showSuccessSnackBar(e.getErrorMessage(context))
+                view?.showErrorSnackBar(e.getErrorMessage(context))
             } finally {
                 view?.showLoading(isLoading = false)
             }
         }
     }
+
+    override fun refresh() {
+        view?.showRefreshing(isRefreshing = true)
+        launch {
+            try {
+                val deposits = solendDepositsInteractor.getAllDeposits(COLLATERAL_ACCOUNTS)
```

Comment:
maybe COLLATERAL_ACCOUNTS - is a domain entity? it's about domain logic, not presentation

Reply:
I think user should know entering presenter what accounts should be visible in that particalar fragment and control it from there. 
So I assume the accounts list is more as a parameter from a user but not as a result of business logic